go/apps/tao_admin/tao_admin.go

```diff
@@ -25,182 +25,239 @@ import (
-	addGuard := flag.Bool("add_guard", false, "Add a trusted guard to the policy")
-	addTPM := flag.Bool("add_tpm", false, "Add trusted platform module to the policy")
-
-	// Flags for the 'user' option, used to create new user keys.
-	userKeyDetails := flag.String("user_key_details", "", "Path to a file that contains an X509Details proto")
-	userKeyPath := flag.String("user_key_path", "usercreds", "key path")
-	userPass := flag.String("user_pass", "", "A password for the new user (for testing only!).")
+var opts = []options.Option{
+	// Flags for all/most commands
+	{"tao_domain", "", "<dir>", "Tao domain configuration directory", "all"},
+	{"quiet", false, "", "Be more quiet", "all"},
+	{"pass", "", "<password>", "Password for policy private key (Testing only!)", "all"},
+
+	// Flags for miscellaneous commands
+	{"config_template", "", "<file>", "Configuration template", "init,newsoft,policy"},
+
+	// Flags for 'newsoft', used to create soft tao keys.
+	{"soft_pass", "", "<pass>", "A password to encrypt the new soft Tao keys", "newsoft"},
+
+	// Flags for 'init'. If these flags are specified, a public cached version
+	// of the domain will also be created.
+	{"pub_domain_address", "", "<adddress>", "Address of TaoCA for public cached domain", "init"},
+	{"pub_domain_network", "tcp", "<network>", "Network of TaoCA for public cached domain", "init"},
+	{"pub_domain_ttl", 30 * time.Second, "<duration>", "Time-to-live of cached policy", "init"},
+
+	// Flags for 'policy' command, used to change and query the policy rules
+	// used for principal authorization. The strings passed to these rules
+	// depend on the Guard given in the domain/tao.
+	{"canexecute", "", "<prog>", "Path of a program to be authorized to execute", "policy"},
+	{"retractcanexecute", "", "<prog>", "Path of a program to retract authorization to execute", "policy"},
+	{"add", "", "<rule>", "A policy rule to be added", "policy"},
+	{"retract", "", "<rule>", "A policy rule to be retracted", "policy"},
+	{"query", "", "<rule>", "A policy query to be checked", "policy"},
+	{"clear", false, "", "Clear all policy rules before other changes", "policy"},
+	{"add_programs", false, "", "Add the program hashes to the policy", "policy"},
+	{"add_containers", false, "", "Add the container hashes to the policy", "policy"},
+	{"add_host", false, "", "Add the host to the policy", "policy"},
+	{"add_vms", false, "", "Add VMs to the policy", "policy"},
+	{"add_linux_host", false, "", "Add LinuxHost to the policy", "policy"},
+	{"add_guard", false, "", "Add a trusted guard to the policy", "policy"},
+	{"add_tpm", false, "", "Add trusted platform module to the policy", "policy"},
+
+	// Flags for 'user' command, used to create new user keys.
+	{"user_key_details", "", "<file>", "File containing an X509Details proto", "user"},
+	{"user_key_path", "usercreds", "<file>", "Key path", "user"},
+	{"user_pass", "", "<pass>", "A password for the new user (for testing only!)", "user"},
```

Comment:
nit: this "for testing only!" warning should match the one on the "pass" option.